src/grid_fields.f90

```diff
@@ -50,7 +50,7 @@ subroutine w_alloc
 
  if(LPf_ord >0)then
   cfl_loc=cfl
-  if(ndim >1)cfl_loc=cfl*yx_rat/sqrt(yx_rat*yx_rat+1.)
+  if(ndim >1) cfl_loc=cfl*yx_rat/sqrt(yx_rat*yx_rat+ndim-1.)
```

Comment:
this one is pretty important. Can you support it?

Reply:
That is *a super important correction* @framira spotted it out and can totally give you update on this.